src/client/debugger/jupyter/debugControllers.ts

```diff
@@ -41,7 +41,7 @@ export class DebugCellController implements DebuggingDelegate {
     }
 }
 
-export class RunByLineController implements DebuggingDelegate {
+export class RunByLineController implements IDebuggingDelegate {
```

Comment:
Curious, do we need all of these interfaces? Why can't we just use classes

Reply:
We were following the extension patterns.